File: LeetCode/1732. Find the Highest Altitude/solution.cpp
/**
 * Runtime: 4 ms
 * Memory Usage: 7.7 MB 
 */
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int ans = 0, pre = 0;
        for (int i = 0; i < gain.size(); ++ i) {
            ans = max(ans, pre + gain[i]);
            pre += gain[i];
        }
        return ans;
    }
};

File: LeetCode/1732. Find the Highest Altitude/solution.py
"""
Runtime: 32 ms
Memory Usage: 14.9 MB
"""
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        ans = pre = 0
        for g in gain:
            ans = max(ans, pre + g)
            pre += g
        return ans
